public/coffee/ide/labels/LabelsManager.coffee

```diff
@@ -0,0 +1,59 @@
+define [
+], () ->
+
+	AUTOMATIC_REFRESH_PERIOD = 1000 * 60 * 10
+
+	class LabelsManager
+		constructor: (@ide, @$scope) ->
+			@$scope.$root._labels = this
+
+			@state =
+				documents: {} # map of DocId => List[Label]
+
+			@loadLabelsTimeout = null
+			@periodicLoadInterval = null
+
+			setTimeout(
```

Comment:
I think its a good idea to avoid these background processes if possible.  We should be able to listen on changes and process the corresponding doc after a delay.  That would catch everything and also work for documents being updated by other users if the event handler is on the document.